src/language-js/print/class.js

```diff
@@ -90,6 +93,22 @@ function printClass(path, options, print) {
   return parts;
 }
 
+const heritageGroupIds = new WeakMap();
+function getHeritageGroupId(node) {
+  if (!heritageGroupIds.has(node)) {
+    heritageGroupIds.set(node, Symbol("heritageGroup"));
+  }
+  return heritageGroupIds.get(node);
+}
+
+const implementsGroupIds = new WeakMap();
+function getImplementsGroupId(node) {
```

Comment:
Can we merge these two functions? Add an extra `description` parameter as  description of `symbol`, I remember there is another similar function in other place.

Reply:
Maybe this PR will be closed(ref: https://github.com/prettier/prettier/issues/10018#issuecomment-762324761). Please look the new PR(#10085) based on this.